src/HealthChecks.SendGrid/SendGridHealthCheck.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using SendGrid;
+using SendGrid.Helpers.Mail;
+
+namespace HealthChecks.SendGrid
+{
+    public class SendGridHealthCheck : IHealthCheck
+    {
+        private const string MailAddressName = "Health Check User";
+        private const string MailAddress = "healthcheck@example.com";
+        private const string Subject = "Checking health is Fun";
+        private readonly string _apiKey;
+        private readonly Func<HttpClient> _httpClientFactory;
+
+        public SendGridHealthCheck(string apiKey, Func<HttpClient> httpClientFactory)
```

Comment:
I know others healthcheck's like ( uris ) using this pattern, but probably inject Ihttpclientfactory is more natural here!!

Reply:
How should I invoke `_httpClientFactory.CreateClient()`
A. with a name or
B. without a name

If A. which value?
I) The name of the healthcheck passed in with `AddSendGrid(...)`
II) A name specific for the `SendGridHealthCheck` class (as in https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/19bb7ed69abe4538a1e0c4e6355a6c24447628ce/src/HealthChecks.UI/Core/Notifications/WebHookFailureNotifier.cs#L35 )